src/vcpkg/vcpkgpaths.cpp

```diff
@@ -620,7 +620,7 @@ namespace vcpkg
 #endif
     }
 
-    VcpkgPaths::VcpkgPaths(Filesystem& filesystem, const VcpkgCmdArguments& args)
+    VcpkgPaths::VcpkgPaths(Filesystem& filesystem, VcpkgCmdArguments& args)
```

Comment:
I'm not a fan of dropping the const here, as it makes `VcpkgPaths` and `VcpkgCmdArguments` circularly coupled (you need to understand both components to understand either). I would prefer a design that just leaves the environment and command line parts separate in the `VcpkgCmdArguments` interface, and compose that here, similar to what I did in https://github.com/microsoft/vcpkg-tool/pull/725 . However, I'm not sure that's a Team opinion rather than "Bill's personal opinion". I think we should talk about it at today's design review meeting since people clearly didn't like what I did either.

Reply:
(To clarify, I think I would do this behavior in `make_manifest_provider` in portfileprovider.cpp)